platforms/core-runtime/tooling-api-provider/src/main/java/org/gradle/tooling/internal/provider/DefaultConnection.java

```diff
@@ -217,7 +216,7 @@ private ProviderOperationParameters validateAndConvert(BuildParameters buildPara
     }
 
     private UnsupportedVersionException unsupportedConnectionException() {
-        return new UnsupportedVersionException(String.format(UNSUPPORTED_MESSAGE, createCurrentVersionMessage()));
+        return new UnsupportedVersionException(String.format("Support for clients using a tooling API version older than 3.0 was removed in Gradle 5.0. %sYou should upgrade your tooling API client to version 3.0 or later.", createCurrentVersionMessage()));
```

Comment:
🤔 since we already change that: should we reuse `MIN_CLIENT_VERSION` here?

Reply:
I'll do that in a follow up PR.